### PR TITLE
updating version, dependencies, and documentation.

### PR DESCRIPTION
**Pull request by DavidGoldwasser**

### CODE REVIEW

**Comment by DavidGoldwasser** [comment]
This CI failure here was expected because if the Ruby dependency upgrade. Tests were passing before this. 

```

Bundler found conflicting requirements for the Ruby version:

  In Gemfile:

    Ruby



    openstudio-extension was resolved to 0.8.0, which depends on

      Ruby (~> 3.2.2)



Ruby (~> 3.2.2), which is required by gem 'openstudio-extension', is not

available in the local ruby installation
```

**Comment by nllong** [comment]
seems like we should bump the minor version of measure-tester-gem too?

**Comment by DavidGoldwasser** [comment]
@nllong if we did that I assume would also update ruby version on gemspec here?
https://github.com/NREL/OpenStudio-measure-tester-gem/blob/develop/openstudio_measure_tester.gemspec#L30

@kflemin? is that ok?

**Comment by kflemin** [comment]
for the new ruby, yes you'll have to bump the minor version on both gems (not the patch level). thanks!

File: openstudio-extension.gemspec [hunk 1]
```diff
@@ -26,16 +26,15 @@ Gem::Specification.new do |spec|
   spec.executables = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.required_ruby_version = '~> 2.7.0'
+  spec.required_ruby_version = '~> 3.2.2'
 
   spec.add_dependency 'bcl', '~> 0.7.1'
   spec.add_dependency 'bundler', '~> 2.1.0'
```
Comment (jmarrec) [inline openstudio-extension.gemspec:32]: bundler and bcl need upgrading too.

cf https://github.com/NREL/openstudio-extension-gem/pull/186/files

**Review by kbenne — APPROVED** [review]

**Pull request merged**